pkg/kubelet/types/pod_update.go

```diff
@@ -202,3 +204,17 @@ func IsRestartableInitContainer(initContainer *v1.Container) bool {
 
 	return *initContainer.RestartPolicy == v1.ContainerRestartPolicyAlways
 }
+
+// HasRestartableInitContainer returns true if the pod has any restartable init
+// container and the feature gate is enabled.
+func HasRestartableInitContainer(pod *v1.Pod) bool {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) {
```

Comment:
Making this include the feature gate check is a little confusing (IsRestartableInitContainer doesn't consider the feature enablement) and could lead to callers thinking a pod doesn't have sidecars when it does.

I think it's better to move the feature gate check to the call sites (there's only three).

Reply:
done.